File: frontend/src/features/projects/api/watchers.ts
import client from '../../../shared/api/client';

export interface TaskWatcher {
  userId: string;
  username: string;
}

export const watchersApi = {
  list: (taskId: string) =>
    client.get<TaskWatcher[]>(`/tasks/${taskId}/watchers`).then((r) => r.data),
  watch: (taskId: string) =>
    client.post(`/tasks/${taskId}/watch`).then((r) => r.data),
  unwatch: (taskId: string) =>
    client.delete(`/tasks/${taskId}/watch`).then((r) => r.data),
};

File: backend/app/api/watchers.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.core.security import get_current_user
from app.models.user import User
from app.services import collaboration_service

router = APIRouter(prefix="/api/tasks/{task_id}", tags=["watchers"])


@router.post("/watch")
async def add_watcher(
    task_id: str,
    user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    collaboration_service.add_watcher(db, user.id, task_id)
    return {"message": "Urmaresti acest task"}


@router.delete("/watch")
async def remove_watcher(
    task_id: str,
    user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    collaboration_service.remove_watcher(db, user.id, task_id)
    return {"message": "Nu mai urmaresti acest task"}


@router.get("/watchers")
async def list_watchers(
    task_id: str,
    user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    return collaboration_service.list_watchers(db, user.id, task_id)
